client/checkout/blocks/index.js

```diff
@@ -34,6 +34,16 @@ const api = new WCPayAPI(
 	request
 );
 
+// Create an API object, which will be used for payment request buttons.
+const paymentRequestApi = new WCPayAPI(
+	{
+		publishableKey: getConfig( 'publishableKey' ),
+		accountId: getConfig( 'accountId' ),
+		locale: getConfig( 'locale' ),
+	},
+	request
+);
```

Comment:
Here in the `blocks` when calling the `registerExpressPaymentMethod`, it was considering the `forceNetworkSavedCards`. But in [the classic checkout](https://github.com/Automattic/woocommerce-payments/blob/develop/client/payment-request/index.js#L33-L38) we don't have the `forceNetworkSavedCards` when registering the payment request button. 
Because of this difference when [this](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L687) has `true` value the express checkout/ payment request button was using different stripe elements (plus payment was failing on the blocks checkout page)

I am still not 200% sure if this change will have other regression scenarios as I not not familiar with the WCPay plugin much. Can you think of any? :grimacing:

Reply:
> I am still not 200% sure if this change will have other regression scenarios as I not not familiar with the WCPay plugin much. 

Cool, I have now learned a lifetime's worth of information about the `forceNetworkSavedCards` flag in a couple separate conversations today. 😅 

In conclusion, I've learned that the following can generally be assumed to be true:
 - When WooPay is disabled, [generally this value will be `false`](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payments.php#L954).
  - The only valid reason for this value to be `true` [is to support WooPay](https://github.com/Automattic/woocommerce-payments/blob/develop/includes/class-wc-payment-gateway-wcpay.php#L687).
   - There is no reason for a Stripe element used to serve a payment request button to ever need to be compatible with WooPay. Moreover, since in this case the payment element will confirm the intent, if we set this value to `true` to create a platform payment method, the request will always fail, because the intent will be on the merchant account. (With WooPay, the intent is confirmed on the WCPay server, where we can clone the created platform payment method to the merchant connect account before confirming the intent.)

Basically, what I'm saying is that I'm pretty confident that this change should be fine and furthermore I can't see a better way to keep both elements functioning as expected without doing what you've done. Sorry for digging up doubt, only to answer my own question. 😹

Reply:
@Mayisha Sorry to reopen this, but I agree with @FangedParakeet's first comment. There's already a parameter inside the `WCPayAPI` to force the Stripe API to be initialized with the merchant account. I think this should work:

1. Passing a parameter `forceAccountRequest` (default to `false`) from `loadStripe` to `getStripe` [here](https://github.com/Automattic/woocommerce-payments/blob/b3d5cf9855cecb72711c6013cadd474c198f5ba8/client/checkout/api/index.js#L101).
2. Passing that parameter as `true` to `api.loadStripe()` [here](https://github.com/Automattic/woocommerce-payments/blob/b6bf28c8a85c0c5c781f2280aaa9cffa1d628dda/client/payment-request/blocks/index.js#L14) and [here](https://github.com/Automattic/woocommerce-payments/blob/b6bf28c8a85c0c5c781f2280aaa9cffa1d628dda/client/payment-request/blocks/index.js#L27).

This should force the blocks payment request button to not use the platform account even though the `forceNetworkSavedCards` is `true`.

Reply:
This worked. Thanks for the suggestion :tada: